aws-janitor/resources/s3_bucket.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"time"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type S3Bucket struct{}
+
+func (S3Bucket) MarkAndSweep(opts Options, set *Set) error {
+	logger := logrus.WithField("options", opts)
+	if !opts.EnableS3BucketClean {
+		logger.Info("Disable s3 bucket clean")
+		return nil
+	}
+	svc := s3.New(opts.Session, &aws.Config{Region: aws.String(opts.Region)})
+	resp, err := svc.ListBuckets(&s3.ListBucketsInput{})
```

Comment:
The ListBucket lists all buckets owned by the authenticated sender of the request([https://github.com/aws/aws-sdk-go/blob/v1.44.137/service/s3/api.go#L6158]), which means that even though we config the region when initializing the client(for example, "us-west-1"), the ListBuckets still returns buckets in all regions, not limited to "us-west-1".
Thus, at line 44, we need to drop buckets which are not in opts.Region, otherwise, the GetBucketTagging will fail since this method only works for buckets with the same region as what we config in client.